File: api/src/test/java/net/blueshell/api/factory/dto/JwtRequestFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.request.JwtRequest;
import org.springframework.stereotype.Component;

/**
 * Factory for JwtRequest test instances.
 */
@Component
@RequiredArgsConstructor
public class JwtRequestFactory extends BaseDtoFactory<JwtRequest> {

    @Override
    public Class<JwtRequest> targetType() {
        return JwtRequest.class;
    }

    @Override
    public JwtRequest createBasic() {
        return new JwtRequest(unique("user"), "Password123!");
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/JwtRequestFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.dto.request.JwtRequest
import org.springframework.stereotype.Component

/**
 * Factory for JwtRequest test instances.
 */
@Component
class JwtRequestFactory : BaseDtoFactory<JwtRequest>() {

    override fun targetType(): Class<JwtRequest> = JwtRequest::class.java

    override fun createBasic(): JwtRequest = JwtRequest(unique("user"), "Password123!")
}
